client/components/workout-detail.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { format } from 'date-fns';
+import Header from './header';
+import JournalPage from '../pages/journal';
+import { exercises, sets } from '../lib/getWorkoutDetails';
+
+class WorkoutDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      exercises: [],
+      date: ''
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  formatDate(date) {
+    const newDate = new Date(date);
+    const formattedDate = format(newDate, 'MMMM dd, yyyy');
+    return formattedDate;
+  }
+
+  componentDidMount() {
+    fetch(`/api/sets/${this.props.workoutId} `)
+      .then(res => res.json())
+      .then(result => this.setState({
+        exercises: sets(exercises(result), result),
+        date: this.formatDate(result[0].workoutDate)
+      }))
+      .catch(err => console.error(err));
+  }
+
+  handleClick() {
+    this.setState({
+      date: ''
+    });
+  }
+
+  getExerciseDetails() {
+    const exercises = this.state.exercises;
+    const fullWorkout = exercises.map((exercise, index) => {
+      const sets = [];
+      for (let i = 0; i < exercise.length; i++) {
+        const set = (
+          <div key={exercise[i].setId} className="row">
+            <div className="col text-white">{i + 1}</div>
+            <div className="col text-white">{exercise[i].reps}</div>
+            <div className="col text-white mb-2">{exercise[i].weight}</div>
+          </div>);
+        sets.push(set);
+      }
```

Comment:
Loop-n-push is _totally_ what `Array.prototype.map` is for. If you need access to the index, `.map()` passes it as the second argument to your callback.

```js
const mapped = array.map((element, index) => {

});
```